File: heaps/heap_min.js
const Heap = require('heap');

// Creating a min heap
const minHeap = new Heap();

// Adding elements to the min heap
minHeap.push(5);
minHeap.push(3);
minHeap.push(8);
minHeap.push(1);

console.log("Min Heap:");
console.log(minHeap.toArray()); // toArray() converts the heap to an array

// Popping elements from the min heap
console.log("Elements popped from min heap:");
console.log(minHeap.pop());
console.log(minHeap.pop());
console.log(minHeap.pop());

console.log("Min Heap after popping:");
console.log(minHeap.toArray());

File: heaps/heap_min.py
import heapq

# Creating a min heap
min_heap = []

# Adding elements to the min heap
heapq.heappush(min_heap, 5)
heapq.heappush(min_heap, 3)
heapq.heappush(min_heap, 8)
heapq.heappush(min_heap, 1)

# You can use heapify to convert a list into a heap in O(n) time
# min_heap = [5, 3, 8, 1]
# heapq.heapify(min_heap)

print("Min Heap:")
print(min_heap)

# Popping elements from the min heap
print("Elements popped from min heap:")
print(heapq.heappop(min_heap))
print(heapq.heappop(min_heap))
print(heapq.heappop(min_heap))

print("Min Heap after popping:")
print(min_heap)
